addons/controllers/clusterbootstrap_controller.go

```diff
@@ -1179,10 +1238,11 @@ func (r *ClusterBootstrapReconciler) updateValuesFromProvider(cluster *clusterap
 			// There are possibilities that current reconciliation loop fails due to various reasons, and during next reconciliation
 			// loop, it is possible that the provider resource has been created. In this case, we want to run update/patch.
 			if apierrors.IsAlreadyExists(err) {
-				// Setting the resource version is because it's been removed at L984 with 'unstructured.RemoveNestedField(newProvider.Object, "metadata")'
```

Comment:
If resource already exists, lets fetch it from the server and only patch ownerReferences. This should cover the use case for Antrea routable pod customization https://github.com/vmware-tanzu/tanzu-framework/issues/1709. 

Patching ownerreferences should be something like https://gist.github.com/vijaykatam/a091168e747872a2176de16f720a6370

Please also add an envtest that pre-creates AntreaConfig.

Reply:
Both ownerrefs and labels.

Reply:
Does that mean during the cluster/tkr upgrade process, GCM should also be responsible for updating the AntreaConfig to new version? So the GCM should watch the TKR associated with the cluster.

Reply:
yes, this also applies to ProviderRef. Above comments have been addressed with the latest commit.